backend/dal/Services/SecurityDepositReturnService.cs

```diff
@@ -0,0 +1,68 @@
+using System.Security.Claims;
+using Microsoft.EntityFrameworkCore;
+using Pims.Dal.Entities;
+using Pims.Dal.Helpers.Extensions;
+using Pims.Dal.Repositories;
+using Pims.Dal.Security;
+
+namespace Pims.Dal.Services
+{
+    public class SecurityDepositReturnService : ISecurityDepositReturnService
+    {
+        readonly ISecurityDepositReturnRepository _securityDepositReturnRepository;
+        readonly ILeaseRepository _leaseRepository;
+        readonly ILeaseService _leaseService;
+        readonly ClaimsPrincipal _user;
+        public SecurityDepositReturnService(ISecurityDepositReturnRepository depositReturnRepository, ILeaseRepository leaseRepository, ILeaseService leaseService, ClaimsPrincipal user)
+        {
+            _securityDepositReturnRepository = depositReturnRepository;
+            _leaseRepository = leaseRepository;
+            _leaseService = leaseService;
+            _user = user;
+        }
+
+        public PimsLease AddLeaseDepositReturn(long leaseId, long leaseRowVersion, PimsSecurityDepositReturn deposit)
+        {
+            ValidateServiceCall(leaseId, leaseRowVersion);
+            deposit.LeaseId = leaseId;
+            _securityDepositReturnRepository.Add(deposit);
+            _securityDepositReturnRepository.CommitTransaction();
+
+            return _leaseRepository.Get(leaseId);
+        }
+
+        public PimsLease UpdateLeaseDepositReturn(long leaseId, long leaseRowVersion, PimsSecurityDepositReturn deposit)
+        {
+            ValidateServiceCall(leaseId, leaseRowVersion);
+            deposit.LeaseId = leaseId;
+            _securityDepositReturnRepository.Update(deposit);
+            _securityDepositReturnRepository.CommitTransaction();
+
+            return _leaseRepository.Get(leaseId);
+        }
+
+        public PimsLease DeleteLeaseDepositReturn(long leaseId, long leaseRowVersion, PimsSecurityDepositReturn deposit)
+        {
+            ValidateServiceCall(leaseId, leaseRowVersion);
+
+            _securityDepositReturnRepository.Delete(deposit.SecurityDepositReturnId);
+            _securityDepositReturnRepository.CommitTransaction();
+
+            return _leaseRepository.Get(leaseId);
+        }
+
+        /// <summary>
+        /// For a deposit service call to be valid, the user must have the lease edit claim and the lease being edited must be up to date.
+        /// </summary>
+        /// <param name="leaseId"></param>
+        /// <param name="leaseRowVersion"></param>
+        private void ValidateServiceCall(long leaseId, long leaseRowVersion)
+        {
+            _user.ThrowIfNotAuthorized(Permissions.LeaseEdit);
```

Comment:
I need to confirm this with the business, but I've heard two different things in meetings now for delete:
1) if a user can edit a lease sub-entity they should be able to delete it.
2) a user should never be able to delete anything unless they have the delete permission.

Reply:
sent an email and cc'd you on it. I've already updated terms with this assumption.

Reply:
Confirmed this with the business. They honestly don't know if the ability to delete needs to be locked down. As a result, we are should control any delete functionality with the delete claim. For now, everyone who has the edit claim will also be given the delete claim, but in the future, that may be more locked down.

Reply:
So yes, any delete functionality should be controlled by the delete claim not the edit claim.

Reply:
@asanchezr

Reply:
@FuriousLlama think this still needs to be updated to have delete controlled by delete claims.